### ISSUE TITLE
Elect candidates in order

### ISSUE THREAD

**Issue by michaelsproul** [michaelsproul/aus_senate#9]
If multiple candidates exceed the quota, then the one with the most votes should be elected first.

**Comment by michaelsproul** [comment]
(this is how it's specified in the legislation)

**Comment by michaelsproul** [comment]
Partly done, just need to deal with bulk exclusions

**Comment by michaelsproul** [comment]
Note: elect all candidates who get a full quota before distributing any of their preferences

**Comment by michaelsproul** [comment]
It seems that this is more complicated than I initially thought. There are several subtleties around when preferences are meant to be transferred according to the legislation, for example: if a candidate is elected on first preferences, then their votes should be transferred and people elected on their preferences should be elected before preferences from any subsequent candidates elected on first preferences have their votes transferred... Basically, the *election* of a candidate needs to be separated from the transfer of their ballots and preferences.

**Comment by michaelsproul** [comment]
Fixed by ce43c3ef6994223b5cd1d0f06b241b06bb2d1660!

For NSW (at least) the election order and vote counts now perfectly match the AEC's:

```
=== Elected ===
Marise PAYNE (Liberal) [1583601 votes]
Sam DASTYARI (Labor) [1385000 votes]
Arthur SINODINOS (Liberal) [1233372 votes]
Jenny McALLISTER (Labor) [1035654 votes]
Fiona NASH (The Nationals) [892890 votes]
Deborah O'NEILL (Labor) [692256 votes]
Concetta FIERRAVANTI-WELLS (Liberal) [547933 votes]
Doug CAMERON (Labor) [351003 votes]
Lee RHIANNON (The Greens) [347579 votes]
John WILLIAMS (The Nationals) [427771 votes]
Brian BURSTON (Pauline Hanson's One Nation) [353829 votes]
David LEYONHJELM (Liberal Democrats) [275766 votes]
```
